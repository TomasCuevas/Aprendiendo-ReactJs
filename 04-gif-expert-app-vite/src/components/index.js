export * from './AddCategory/AddCategory.jsx';
export * from './GifGrid/GifGrid.jsx';
export * from './GifCard/GifCard.jsx';