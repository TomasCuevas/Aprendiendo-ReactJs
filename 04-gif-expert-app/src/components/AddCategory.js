import React, { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState( 'Hola Mundo' );
    
    const handleInputChange = ( e ) => {

        setInputValue( e.target.value );
    }

    const handleSubmit = ( e ) => {
        
        e.preventDefault();
    }

    return (
        <>
            <h2>Agregar Categoria</h2>
            <form onSubmit={ handleSubmit } >
                <input 
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
            </form>
        </>
    )
}
