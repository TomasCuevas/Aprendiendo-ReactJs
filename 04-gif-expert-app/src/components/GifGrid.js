import React from 'react'

export const GifGrid = ({ category }) => {
	
	const getGif = async() => {

		const url = 'http://api.giphy.com/v1/gifs/search?q=rick%20and%20morty&limit=10&api_key=qV5k5fNfElMQkPpD5DWCX7fVs5zPoHYe'

		const res = await fetch( url )
		const { data } = await res.json();

		const gifs = data.map( ( img ) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images.downsized_medium.url
			}
		})
		
		console.log( gifs );
	}

	getGif();

	return (
		<div>
			<h3>{ category }</h3>
		</div>
	)
}
