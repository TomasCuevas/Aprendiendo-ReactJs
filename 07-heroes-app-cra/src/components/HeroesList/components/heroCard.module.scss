.card__link {
  max-width: 22rem;
  min-height: 35rem;
  max-height: 35rem;
  margin: 1rem .5rem;
  border-radius: 2rem;
  overflow: hidden;
  text-decoration: none;
  animation-name: card;
  animation-duration: 1.3s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  
  .list__element {
    width: 100%;
    height: 100%;
    list-style: none;

    .card__container {
      display: grid;
      grid-template-rows: 80% 20%;
      align-items: center;
      width: 100%;
      height: 100%;

      .card__img {
        width: 100%;
        height: 100%;
        object-fit: cover;
      }

      .card__info {
        align-items: center;
        border-bottom: 2px solid darkcyan;
        border-left: 2px solid darkcyan;
        border-radius: 0 0 2rem 2rem;
        border-right: 2px solid darkcyan;
        display: flex;
        flex-direction: column;
        justify-content: center;
        height: 100%;
        transition: .3s ease-in;
        user-select: none;
        width: 100%;
        text-decoration: none;

        .card__superhero {
          margin: .2rem 0;
          color: #000;
          font-weight: 300;
          font-size: 1.5rem;
          transition: .3s ease-in;
          text-decoration: none;
        }
        
        .card__publisher {
          color: #000;
          transition: .5s ease-in;
          text-decoration: none;
        }
      }
      
    }
    .card__container:hover .card__info {
      background-color: darkcyan;
    }
    .card__container:hover .card__superhero {
      letter-spacing: 5px;
      color: #FFF;
    }
    .card__container:hover .card__publisher {
      letter-spacing: 3px;
      word-spacing: 5px;
      color: #FFF;
      text-decoration: underline;
      text-underline-offset: 5px;
    }
    .card__container:hover .card__link {
      letter-spacing: 5px;
    }
  }
}

@keyframes card {
  0% {
    opacity: 0;
    transform: scale(1.2);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}