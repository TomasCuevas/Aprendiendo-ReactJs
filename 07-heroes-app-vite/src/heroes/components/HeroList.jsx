import { getHeroesByPublisher } from '../helpers';

export const HeroList = ({ publisher }) => {

  const heroes = getHeroesByPublisher( publisher );

  return (
    <>
      {
        heroes.map((hero) => (
          <li key={hero.id}>
            { hero.superhero }
          </li>
        ))
      }
    </>
  )
}