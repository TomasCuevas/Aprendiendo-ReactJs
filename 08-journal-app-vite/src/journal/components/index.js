export * from './ImageGalery'
export * from './Navbar'
export * from './Sidebar'