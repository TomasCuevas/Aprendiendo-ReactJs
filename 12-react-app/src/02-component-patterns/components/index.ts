import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductCardHOCProps } from '../interfaces/ProductInterfaces';

import { ProductButtons } from './ProductButtons';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';

export * from './ProductButtons';
export * from './ProductImage';
export * from './ProductTitle';

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Buttons: ProductButtons,
  Image: ProductImage,
  Title: ProductTitle,
});
