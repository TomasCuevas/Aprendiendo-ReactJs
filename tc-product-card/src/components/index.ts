import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductCardHOCProps } from '../interfaces/ProductInterfaces';

import { ProductButtons } from './ProductButtons';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';

export { ProductButtons, ProductImage, ProductTitle };

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Buttons: ProductButtons,
  Image: ProductImage,
  Title: ProductTitle,
});
