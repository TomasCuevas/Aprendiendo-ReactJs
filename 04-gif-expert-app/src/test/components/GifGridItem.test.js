import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe( 'Prueba en <GridGifItem>', () => {

    test( 'Hacer snapshot de el componente', () => {

        const wrapper = shallow( <GifGridItem /> )

        expect( wrapper ).toMatchSnapshot();
    })
})