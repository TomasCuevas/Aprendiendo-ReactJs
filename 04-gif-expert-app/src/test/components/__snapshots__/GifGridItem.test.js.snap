// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GridGifItem> Hacer snapshot de el componente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img />
  <p />
</div>
`;
